app/components/buttons/ModalButton.js

```diff
@@ -24,6 +25,17 @@ class ModalButton extends React.Component {
     onSubmit && this.props.onSubmit(...args);
   }
 
+  onShow() {
+    const { onShow } = this.props;
+    onShow && this.props.onShow();
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.showModal) {
```

Comment:
Probably you want to change this to `if (nextProps.showModal && !this.props.modal)` so that modal showing is triggered only when the showModal props is changing. Otherwise, any other prop changing can trigger the modal to be shown again while showModal is true.

Reply:
While i wasnt able to reproduce the error, changes have been effected where errors were found.